src/services/photo/index.js

```diff
@@ -49,73 +68,122 @@ function sendUploadToGCS (req, res, next) {
   stream.end(req.file.buffer);
 }
 
+function uploadToGCSPromise(reqFile) {
+  return new Promise(function (resolve, reject) {
+    if (!reqFile) {
+      return reject(new Error('No file provided'));
+    }
+
+    const gcsname = Date.now() + '_' + reqFile.originalname;
+    const bucketFile = bucket.file(gcsname);
+    const stream = bucketFile.createWriteStream();
+
+    stream.on('error', function (err) {
+      reqFile.cloudStorageError = err;
+
+      return reject(err);
+    });
+
+    stream.on('finish', function () {
+      reqFile.cloudStorageObject = gcsname;
+      reqFile.cloudStoragePublicUrl = getPublicUrl(gcsname);
+
+      return resolve();
+    });
+
+    stream.end(reqFile.buffer);
+  });
+}
+
+function respondWithImageMetadata(reqFile) {
+  return new Promise(function (resolve, reject) {
+    // we don't care req.body for now.
+    // (care only file content)
+    var imageUrl;
+
+    if (reqFile && reqFile.cloudStoragePublicUrl) {
+      imageUrl = reqFile.cloudStoragePublicUrl;
+    }
+
+    return resolve({
+      url: imageUrl,
+      mimetype: reqFile.mimetype,
+      size: reqFile.size
+    });
+  });
+}
+
+class PhotosService {
+  create(data, params) {
+    uploadToGCSPromise(data.file)
+    .then(() => {
+      return respondWithImageMetadata(data.file);
+    })
+    .catch(function (err) {
+      return Promise.reject(err);
+    });
+  }
+}
+
+class UploadPhotoFromUrlService {
+  create(data, params) {
+    if (typeof data !== 'object') {
```

Comment:
Why do we need this?
I think it is checked already in feathers middleware as I pointed out.

https://github.com/feathersjs/feathers-commons/blob/master/src/arguments.js#L66